mongo/crud_examples_test.go

```diff
@@ -0,0 +1,128 @@
+// Copyright (C) MongoDB, Inc. 2017-present.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License. You may obtain
+// a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+package mongo_test
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"time"
+
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+)
+
+// Client examples
+
+func ExampleClient_ListDatabases() {
+	var client *mongo.Client
+
+	// use a filter to only select the admin database
+	// specify the NameOnly option so the returned specificiations only have the name field populated
+	opts := options.ListDatabases().SetNameOnly(true)
+	result, err := client.ListDatabases(context.TODO(), bson.D{{"name", "admin"}}, opts)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	for _, db := range result.Databases {
+		fmt.Println(db.Name)
+	}
+}
+
+func ExampleClient_ListDatabaseNames() {
+	var client *mongo.Client
+
+	// use a filter to only select the admin database
+	result, err := client.ListDatabaseNames(context.TODO(), bson.D{{"name", "admin"}})
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	for _, db := range result {
+		fmt.Println(db)
+	}
+}
+
+func ExampleClient_Watch() {
+	var client *mongo.Client
+
+	// specify a pipeline that will add a field with the key "newField" to each event returned
+	// specify the MaxAwaitTimeOption to have each attempt wait two seconds for new documents
+	addFieldsStage := bson.D{{"$addFields", bson.D{{"newField", "this is a new field"}}}}
```

Comment:
Nit: consider something more useful, like a $match that includes only insert events

Reply:
done.